api/WebApi.ts

```diff
@@ -42,8 +42,7 @@ import os = require('os');
 import url = require('url');
 import path = require('path');
 
-const isBrowser: boolean =(function(){return typeof window !== 'undefined' && this===window})();
-
+const isBrowser: boolean =(function(){return typeof window !== 'undefined' && typeof window.document !== 'undefined'})();
```

Comment:
Is `window` ever defined outside of the context of a browser? Aka can we simplify this to:
`const isBrowser: boolean = typeof window !== 'undefined'`

Reply:
window should not be accessible on the server side, so yes, we should only need the first check.

Reply:
I also don't think we need to wrap it in a function (and didn't need to be doing so initially either), so it can just be `const isBrowser: boolean = typeof window !== 'undefined'`